server/src/main/java/org/apache/druid/server/coordinator/LoadQueuePeon.java

```diff
@@ -21,37 +21,48 @@
 
 import org.apache.druid.timeline.DataSegment;
 
+import java.util.Map;
 import java.util.Set;
 
 /**
  * This interface exists only to support configurable load queue management via curator or http. Once HttpLoadQueuePeon
  * has been verified enough in production, CuratorLoadQueuePeon and this interface would be removed.
  */
 @Deprecated
-public abstract class LoadQueuePeon
+public interface LoadQueuePeon
```

Comment:
Change summary:
Added methods to `cancelLoad`, `cancelDrop`, and `loadSegment(segment, type)` (primary, replica or balance)